pkg/node/manager/manager.go

```diff
@@ -411,7 +411,8 @@ func (m *Manager) NodeUpdated(n nodeTypes.Node) {
 			tunnelIP = nodeIP
 		}
 
-		if address.Type == addressing.NodeInternalIP {
+		if option.Config.IptablesMasqueradingEnabled() &&
+			address.Type != addressing.NodeInternalIP {
```

Comment:
In the original PR this check was `address.Type != addressing.NodeCiliumInternalIP {` :thinking:

Reply:
Yes, this looks fine.